src/hotspot/share/prims/wbtestmethods/parserTests.cpp

```diff
@@ -127,6 +127,14 @@ static void fill_in_parser(DCmdParser* parser, oop argument)
      } else {
       parser->add_dcmd_option(argument);
      }
+   } else if (strcmp(type, "FILE") == 0) {
+     DCmdArgument<FileArgument> *argument =
+      new DCmdArgument<FileArgument>(name, desc, "FILE", mandatory);
```

Comment:
On top: We hug the `*`s next to the type in Hotspot, not next to the var name. So `DCmdArgument<FileArgument>* argument`. This is something to check for all new code.

Pre-existing: The indentation of the if-block is wrong.

Also, @SoniaZaldana, would you mind changing the code to this (does not include your change), the repetition just made me cringe 😅.

```c++
  DCmdArgument<char*>* argument = nullptr;
  if (strcmp(type, "STRING") == 0) {
    argument = new DCmdArgument<char*>(name, desc, "STRING", mandatory, default_value);
  } else if (strcmp(type, "NANOTIME") == 0) {
    DCmdArgument<NanoTimeArgument>* argument = new DCmdArgument<NanoTimeArgument>(name, desc, "NANOTIME", mandatory, default_value);
  } else if (strcmp(type, "JLONG") == 0) {
    DCmdArgument<jlong>* argument = new DCmdArgument<jlong>(name, desc, "JLONG", mandatory, default_value);
  } else if (strcmp(type, "BOOLEAN") == 0) {
    DCmdArgument<bool>* argument = new DCmdArgument<bool>(name, desc, "BOOLEAN", mandatory, default_value);
  } else if (strcmp(type, "MEMORYSIZE") == 0) {
    DCmdArgument<MemorySizeArgument>* argument = new DCmdArgument<MemorySizeArgument>(name, desc, "MEMORY SIZE", mandatory, default_value);
  } else if (strcmp(type, "STRINGARRAY") == 0) {
    DCmdArgument<StringArrayArgument*>* argument = new DCmdArgument<StringArrayArgument*>(name, desc, "STRING SET", mandatory);
  }

  if (argument != nullptr) {
    if (isarg) {
      parser->add_dcmd_argument(argument);
    } else {
      parser->add_dcmd_option(argument);
    }
  }
```

Reply:
@jdksjolen 

> Also, @SoniaZaldana, would you mind changing the code to this 

Even simpler (did not test, but you get my drift):

```
#define ALL_TYPES_DO_XX(what) \
  what(char*, "STRING") \
  what(NanoTimeArgument, NANOTIME) \
  what(jlong, "JLONG") 
... etc
```
then

```
#define XX(TYPE, NAME) \
if (strcmp(type, NAME) == 0) { \
    DCmdArgument<TYPE>* argument = new DCmdArgument<TYPE>(name, desc, NAME, mandatory, mandatory, default_value); \
}
ALL_TYPES_DO_XX(XX)
#undef XX
```

;-)

Reply:
Sonia, my bad if you already know this stuff but since it's fairly esoteric knowledge nowadays I'd like to help you out in advance: Thomas is proposing the usage of a X macro https://en.wikipedia.org/wiki/X_macro

These can be found throughout Hotspot, you can find an example definition and usage in `logTag.hpp` and `logTag.cpp`.

Reply:
@SoniaZaldana Note that this is very much optional.

Reply:
Hi folks, thanks for the pointers! I wasn't familiar with X macros and after some time toying around with them, I'm sad to report that I am not a fan (yet!).

I implemented it and ended up breaking part of the tests. I quickly realized that debugging these is a bit harder for less experienced c++ developers (like myself). 

So, just wanted to note: 
- I cleaned up the indentation in this function as it was all wrong. 
- I didn't get rid of the repetition. Tried to but quickly realized we can't pull the DCmdArgument out of the if statements as they're different types. 

And note to self, to keep reviewing X macros because they did shorten the code a lot when I implemented them. Perhaps I'll give it another go in a different RFE.

Sorry it's not what either of you hoped for!

Reply:
That's fine, thanks for having a go!